core/genesis.go

```diff
@@ -177,17 +178,29 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *Genesis, override
 		}
 		return genesis.Config, block.Hash(), nil
 	}
+
 	// We have the genesis block in database(perhaps in ancient database)
 	// but the corresponding state is missing.
 	header := rawdb.ReadHeader(db, stored, 0)
-	var cfg *trie.Config = nil
-	if genesis.Config.UseVerkle {
-		cfg = &trie.Config{UseVerkle: true}
-	}
-	if _, err := state.New(header.Root, state.NewDatabaseWithConfig(db, cfg), nil); err != nil {
-		if genesis == nil {
-			genesis = DefaultGenesisBlock()
+
+	var trieCfg *trie.Config
+
+	if genesis == nil {
+		storedcfg := rawdb.ReadChainConfig(db, stored)
+		if storedcfg == nil {
+			panic("this should never be reached: if genesis is nil, the config is already present or 'geth init' is being called which will create it")
```

Comment:
I haven't tried running this yet, however this message looks weird to me: if `geth init` is being called, surely it shouldn't panic, right?

Reply:
It's a situation that can't/shouldn't be reached.  Hence the panic.

There are two cases where the enclosing `if` on L188 is `true`:
    * db has no genesis
        * `genesis != nil` (set above, and returns) so this won't be reached
    * db has genesis
        * config should have been stored along with genesis so this won't be reached

Reply:
I changed the message wording to reflect this.